### PR TITLE
Implement Exit command and add command validator

### PR DESCRIPTION
**Pull request by 1AmNegan**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### Description of the Change

* Command validator checks if the command has been called in the right location and if the session is valid (2 hours expiration) and gives helpful output.
* Exit command first calls the command validator and if everyting is fine, it deletes the stored preferences.

### CODE REVIEW

File: lib/utils/preference-manager.js [hunk 1]
```diff
@@ -3,7 +3,13 @@ const prefDirectory = `${require('os').homedir()}/.autolabjs`;
 
 const cliConfig = new Preferences('autolabjs.cli.config', {}, {
   encrypt: false,
-  file: `${prefDirectory}/cli-config.json`,
+  file: `${prefDirectory}/prefs.json`,
```
Comment (prasadtalasila) [inline lib/utils/preference-manager.js]: One observation. The preference-manager.js module has too many elementary getter / setter functions. Is it better to organize the work around manipulating preferences, credentials, locations json structures? Such a set of json objects and the methods would be more easy to reason about in the domain of AutolabJS.

File: test/feature/features/exit.feature [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Feature: Exit
+  I should logout of AutolabJS when I run exit command
+
+  Scenario: Exit command
+    Given I have already logged in
+    When I run exit command
+    Then My login credentials should be removed
```
Comment (prasadtalasila) [inline test/feature/features/exit.feature]: other useful scenarios are:
1. execution of `autolabjs exit` command twice in correct location containing AutolabJS assignment.
1. execution of `autolabjs exit` in any random location not containing AutolabJS assignment.

File: lib/utils/preference-manager.js (hunk 1 above)
Comment (prasadtalasila) [reply lib/utils/preference-manager.js]: Also the code coverage for preference-manager.js needs to grow towards complete coverage.

File: test/integration/exit.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+const chai = require('chai');
```
Comment (prasadtalasila) [inline test/integration/exit.js]: please modify the integration test for `autolabjs init` command. As can be seen from [travis log](), there is a leakage of test-specific information to console. This leak must be prevented.

Comment (1AmNegan) [reply test/integration/exit.js]: This is the output of the *inquirer*, which is not using ``console.log``. I need to look at the source code and see how it is managing the output.

File: lib/utils/preference-manager.js (hunk 1 above)
Comment (1AmNegan) [reply lib/utils/preference-manager.js]: There is another way to handle preferences - 
There will be one getter and one setter
```Javascript
setPreference({
  name: 'preference_name',
  value: {
    option1: 'option1_value',
    option2: 'option2_value'
  }
});
```
And similarly for getter. Is this approach fine?

File: test/integration/exit.js (hunk 3 above)
Comment (prasadtalasila) [reply test/integration/exit.js]: Can we mock inquirer itself?

File: lib/utils/preference-manager.js (hunk 1 above)
Comment (prasadtalasila) [reply lib/utils/preference-manager.js]: Yup. Sounds good and is compact too.

The typical way to handle this is to retrieve the existing preferences, add/modify the needed values and save them again.

File: test/integration/exit.js (hunk 3 above)
Comment (1AmNegan) [reply test/integration/exit.js]: With sinon, we cannot do it. This is because sinon is entirely function-based.

**Pull request merged**